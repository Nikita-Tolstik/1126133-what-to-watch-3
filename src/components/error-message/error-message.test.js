import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from './error-message.jsx';
import {Provider} from 'react-redux';
import configureStore from "redux-mock-store";
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

it(`Render ErrorMessage - without error`, () => {

  const store = mockStore({
    [NameSpace.DATA]: {
      isError: false
    },
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ErrorMessage />
      </Provider>
  ).toJSON();


  expect(tree).toMatchSnapshot();
});

it(`Render ErrorMessage - with error`, () => {

  const store = mockStore({
    [NameSpace.DATA]: {
      isError: true
    },
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ErrorMessage />
      </Provider>
  ).toJSON();


  expect(tree).toMatchSnapshot();
});
