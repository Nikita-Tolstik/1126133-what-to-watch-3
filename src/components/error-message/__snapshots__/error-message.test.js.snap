// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ErrorMessage - with error 1`] = `
<div
  className="error"
>
  <p
    className="error__message"
  >
    Internal Server Error. Please, refresh page.
  </p>
  <button
    className="error__button"
    href="#"
    onClick={[Function]}
  >
    Refresh page
  </button>
</div>
`;

exports[`Render ErrorMessage - without error 1`] = `null`;
